gpflow/utilities/utilities.py

```diff
@@ -115,7 +117,10 @@ def leaf_components(input: tf.Module):
 
 def _merge_leaf_components(
         input: Dict[str, Union[tf.Tensor, Parameter]]) -> Dict[str, Union[tf.Tensor, Parameter]]:
-    if len(set(input.values())) == len(input):
+    input_values = set(
+        [value.experimental_ref() if isinstance(value, tf.Variable) else value for value in input.values()]
```

Comment:
what does this change do? (not mentioned in PR summary)

Reply:
tf.Variables are not hashable, so you cannot take set of a list of them. Instead, tensorflow recommends to use  .experimental_ref(). You can replicate it yourself if you want :)